### PR TITLE
MyBatis, Model, and WebService updates based on fialized schema

### PR DESCRIPTION
**Pull request by zmoore-usgs**

### CODE REVIEW

File: mlr-data-postgres-impl/src/main/resources/mybatis/mybatis.conf.xml [hunk 1]
```diff
@@ -15,7 +15,18 @@
 	</settings-->
 
 	<typeAliases>
+		<typeAlias alias="Agency" type="gov.usgs.wma.model.Agency"/>
+		<typeAlias alias="ErrorModel" type="gov.usgs.wma.model.ErrorModel"/>
+		<typeAlias alias="Feature" type="gov.usgs.wma.model.Feature"/>
+		<typeAlias alias="GeoCodeBase" type="gov.usgs.wma.model.GeoCodeBase"/>
+		<typeAlias alias="Geometry" type="gov.usgs.wma.model.Geometry"/>
```
Comment (kmschoep-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/mybatis.conf.xml]: There won't be Geometry as such in the database, the geometry definition will be part of the geoJSON that will be stored in the feature column.

File: mlr-data-postgres-impl/src/main/resources/mybatis/mybatis.conf.xml [hunk 2]
```diff
@@ -15,7 +15,18 @@
...
 		<typeAlias alias="MonitoringLocation" type="gov.usgs.wma.model.MonitoringLocation"/>
+		<typeAlias alias="MonitoringLocationAquifer" type="gov.usgs.wma.model.MonitoringLocationAquifer"/>
+		<typeAlias alias="MonitoringLocationType" type="gov.usgs.wma.model.MonitoringLocationType"/>
+		<typeAlias alias="Point" type="gov.usgs.wma.model.Point"/>
```
Comment (kmschoep-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/mybatis.conf.xml]: There won't be Point as such in the database, the geometry definition will be part of the geoJSON that will be stored in the feature column.

File: mlr-data-postgres-impl/src/main/resources/mybatis/mybatis.conf.xml [hunk 3]
```diff
@@ -15,7 +15,18 @@
...
+		<typeAlias alias="PointProperties" type="gov.usgs.wma.model.PointProperties"/>
```
Comment (kmschoep-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/mybatis.conf.xml]: There won't be PointProperties as such in the database, the geometry definition will be part of the geoJSON that will be stored in the feature column.

File: mlr-data-postgres-impl/src/main/resources/mybatis/mybatis.conf.xml [hunk 4]
```diff
@@ -15,7 +15,18 @@
...
+		<typeAlias alias="Point3D" type="gov.usgs.wma.model.Point3d"/>
```
Comment (kmschoep-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/mybatis.conf.xml]: There won't be Point3D as such in the database, the geometry definition will be part of the geoJSON that will be stored in the feature column.

**Comment by kmschoep-usgs** [comment]
Added comments about the definitions inside the feature model.  Maybe doesn't hurt to have them stubbed out if we change directions, but for now those are encapsulated in the geoJSON that will be inserted into monitoring_location.feature column.

File: mlr-data-postgres-impl/src/main/resources/mybatis/MonitoringLocationMapper.xml [hunk 5]
```diff
@@ -69,23 +93,29 @@
...
 		<result property="name" column="name"/>
 		<result property="code" column="code"/>
 	</resultMap>
 	
 	<!-- Map Uses-->
 	<resultMap id="UseMap" type="Use">
-		<id property="id" column="id"/>
+		<id property="id" column="use_id"/>
+		<!--use_type from DB not in Model?-->
 		<result property="rank" column="rank"/>
```
Comment (mhines-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/MonitoringLocationMapper.xml]: The use table got changed a bit too, it's now 
use_id integer
code text
name text
use_type text

**Comment by zmoore-usgs** [comment]
@kmschoep-usgs 

Gotcha, makes sense to me. One thing I wasn't sure about at this point is how to map the feature column on the Java side. I'm not sure how MyBatis handles json types. 

It sounds like we'll need to make some updates to the Java model classes to incorporate the shift to GeoJSON.

**Comment by kmschoep-usgs** [comment]
@zmoore-usgs , I don't think mybatis will "care" about the data types.  It'll be learning how to query in postgres (https://www.postgresql.org/docs/9.6/static/datatype-json.html#JSON-KEYS-ELEMENTS).  I'm guessing it turns into something like a string when it's queries and it can then be parsed for its elements from there in the java layer.

File: mlr-data-postgres-impl/src/main/resources/mybatis/MonitoringLocationMapper.xml (hunk 5 above)
Comment (zmoore-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/MonitoringLocationMapper.xml]: Yeah this table was a little confusing to me. It seems the rank is now stored in the relational table for uses? The generated Java model for Use doesn't have a UseType field so we will need to update that

Comment (kmschoep-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/MonitoringLocationMapper.xml]: yep, and that table (monitoring_location_use_rel) will need to be added as a model as well.

File: mlr-data-postgres-impl/src/main/resources/mybatis/MonitoringLocationMapper.xml [hunk 6]
```diff
@@ -102,10 +106,72 @@
...
+	
+	<select id="getMonitoringLocation_Uses" resultMap="UseMap">
+		left outer join mlr_data.monitoring_location_use_rel MUR 
+			on MUR.monitoring_location_id = ML.monitoring_location_id
+			left outer join mlr_data.use US
+				on MUR.use_id = US.use_id
+	</select>
+	
+	<select id="getMonitoringLocation_Agencies" resultMap="UseMap">
+		left outer join mlr_data.monitoring_location_use_rel MUR
```
Comment (kmschoep-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/MonitoringLocationMapper.xml]: copy pasta--need to substitute 'use' with 'agency'.

Comment (zmoore-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/MonitoringLocationMapper.xml]: Ah yeah I copied that to add agency fetching right at the end of the day Friday but didn't actually update it and didn't mean to commit it yet with the rest of this. Whoops 😝

**Comment by zmoore-usgs** [comment]
@thongsav-usgs @dpattermann-usgs  Going to go ahead and merge this ahead of demo. Should probably be reviewed post-merge though.

**Pull request merged**